### PR TITLE
Add config variable checks

### PR DESCRIPTION
**Pull request by superkhau**
- Check for config variables defined as functions
- Check for config variables that are undefined

### CODE REVIEW

**Comment by superkhau** [comment]
@raymondfeng Review please.

File: test/executor.test.js [hunk 1]
```diff
@@ -494,6 +493,37 @@ describe('executor', function() {
       });
     });
 
+    it('should not parse config variables in functions',
+        function(done) {
+      var config =  simpleMiddlewareConfig('routes', {
+        path: Function
```
Comment (raymondfeng) [inline test/executor.test.js]: This is confusing. It should be something like `path: function() {}`.

**Comment by raymondfeng** [comment]
Can you also add a testcase to cover an array of string/non-string values?

**Comment by superkhau** [comment]
LGTM by @raymondfeng over hangout.

**Pull request merged**